src/server/scene/basic_surface.cpp

```diff
@@ -263,16 +263,7 @@ ms::BasicSurface::BasicSurface(
     cursor_stream_adapter{std::make_unique<ms::CursorStreamImageAdapter>(*this)},
     session_{session}
 {
-    auto callback = [this, observers=weak(observers)](auto const& size)
-        {
-            if (auto const o = observers.lock())
-                o->frame_posted(this, 1, size);
-        };
-
-    for (auto& layer : layers)
-    {
-        layer.stream->set_frame_posted_callback(callback);
-    }
+    update_frame_posted_callbacks(ProofOfMutexLock::Fake{});
```

Comment:
OK now I'm confused. Why do we require `ProofOfMutexLock` if we fake it in production code?

We need at least a comment explaining or (better) code that makes the conditions clear.

Reply:
I will try to improve names and documentation around this